src/cpu/aarch64/cpu_isa_traits.hpp

```diff
@@ -194,6 +197,7 @@ static inline bool mayiuse(const cpu_isa_t cpu_isa, bool soft = false) {
         case sve_512:
             return cpu().has(XBYAK_AARCH64_HWCAP_SVE)
                     && cpu().getSveLen() >= SVE_512;
+        case feat_bf16: return getauxval(AT_HWCAP2) & (1UL << 14);
```

Comment:
(nit): Can it be backported to Aarch64 Xbyak so we can use something like `cpu().has(XBYAK_AARCH64_HWCAP_BF16)` so all logic lives in one place?

Reply:
@milpuz01 
Thank you for implementing BF16 feature detection. It's very nice Arm's BF16 is supported.

I modified your implementation a bit just to be similar to other implementations.  Could you please check that
https://github.com/kawakami-k/oneDNN/commits/aarch64_jit__reorder_fp32_bf16_xbyak
is suitable for you? If it is, could you take the change into this PR? (My name does not have to be in the commit log of this PR).

Xbyak_aarch64 files in oneDNN are just copied from https://github.com/fujitsu/xbyak_aarch64 
I will update the change into https://github.com/fujitsu/xbyak_aarch64 as well.

Reply:
@milpuz01 Could you please update the PR by utilizing XByak API for bf16 detection?

Reply:
@milpuz01 @nSircombe Any updates on this PR?

Reply:
Hi @igorsafo and @kawakami-k, in @kawakami-k PR https://github.com/oneapi-src/oneDNN/pull/1595 there is check for availability of macro `HWCAP2_BF16` (see here: https://github.com/oneapi-src/oneDNN/blob/master/src/cpu/aarch64/xbyak_aarch64/src/util_impl_linux.h#L394) which is not available in all kernels that might run on Neoverse V1 platforms that have support for BF16. For example, if you pick AWS Graviton 3 instance you can run Ubuntu 20.04 which has kernel 5.15.0 which doesn't have `HWCAP2_BF16` so that macro cannot be used and bit position needs to be manually specified as with what we did in this PR. Thus, using `mayiuse_bf16()` as suggested by @kawakami-k would fail to return `true`.

Can we because of this either accept approach laid out in this PR or @kawakami-k do you think that you can avoid using `HWCAP2_BF16` in your approach and align it with reading `AT_HWCAP2` register using `getauxval()` and checking whether 14th bit is set?

Reply:
@milpuz01 @kawakami-k How bad would it be to have something like this? I can't think of a case when we break anything for the current HW support.
```cpp
#ifndef HWCAP2_BF16
#define HWCAP2_BF16 (1UL << 14)
#endif
if (hwcap & HWCAP2_BF16)
    type_ |= (Type)XBYAK_AARCH64_HWCAP_BF16;
#undef HWCAP2_BF16
...

```

Reply:
> If it's good to you, I'll also port it to "https://github.com/fujitsu/xbyak_aarch64".

Hi @kawakami-k , thanks for your quick response. Yes, it does sound good to me and I agree that it should be in Xbyak_aarch64. Please make sure that when HWCAP2_BF16 is not defined register that needs to be read is AT_HWCAP2 and not AT_HWCAP

Reply:
@milpuz01 @kawakami-k There is a small bug in my code. We probably don't want to undef `HWCAP2_BF16` if it was defined prior to our code. Otherwise any code that includes `xbyak_aarch64` will lose `HWCAP2_BF16`

Reply:
@milpuz01 
Could you try this patch?
https://github.com/kawakami-k/oneDNN/commit/9a9430c7db870b78c6402d786a67921af4a66334

If it's good to you, could you cherry-pick it to this PR?

Reply:
@kawakami-k, thanks for the patch. I can confirm that it works, but I cannot cherry-pick to this PR. Would you mind creating PR for your patch please? Once there is PR I would be able then to cherry-pick. Thanks.

Reply:
@milpuz01 
Could you take the change in `util_impl_linux.h` https://github.com/kawakami-k/oneDNN/commit/9a9430c7db870b78c6402d786a67921af4a66334 directly into your commit? It's no problem the change is committed by your side :-).

Reply:
HI @kawakami-k, as that change is on your private branch https://github.com/kawakami-k/oneDNN/tree/check_bf16 I can't import it to this commit? Would you mind just raising PR for it (https://github.com/oneapi-src/oneDNN/compare/master...kawakami-k:oneDNN:check_bf16) and once that PR is raised I would be able then to add it to this commit?

Many thanks,
Milos.

Reply:
@milpuz01 
Sorry for inconvenience. I did it. Thanks!

Reply:
Thanks @kawakami-k, I have imported it now.